apps/web/src/pages/admin/users/index.tsx

```diff
@@ -0,0 +1,215 @@
+import debounce from 'lodash.debounce'
+import { GetServerSideProps } from 'next'
+import useTranslation from 'next-translate/useTranslation'
+import { useCallback, useMemo, useState } from 'react'
+
+import Breadcrumbs from '@/components/breadcrumbs'
+import Checkbox from '@/components/checkbox'
+import Pagination from '@/components/pagination/pagination'
+import Panel from '@/components/panel'
+import Table from '@/components/table'
+import { ColumnDefinitionType } from '@/components/table'
+import TextInput from '@/components/text-input/text-input'
+import usePagination from '@/hooks/use-pagination'
+import AdminLayout from '@/layouts/admin-layout'
+import adminService from '@/services/admin-service'
+import { isAuthenticatedUserAdmin } from '@/services/api/auth-service'
+import { getUsersFilterQuery } from '@/utils/filters'
+import { useAuthApi } from '@/utils/swr'
+import { urls } from '@/utils/urls'
+
+const USERS_PER_PAGE = 10
+
+interface Role {
+  id: FirebaseClaim
+  label: string
+}
+
+interface RendererProps {
+  value: any
+  item: UserAccount
+  colKey: string
+}
+
+interface CheckRoleProps {
+  user: UserAccount
+  role: Role
+}
+
+function CheckRole({ user, role }: CheckRoleProps) {
+  const [checked, setChecked] = useState(user.claims.includes(role.id))
+
+  const updateClaim = useCallback(async () => {
+    // show in UI before it goes through
+    setChecked(!checked)
+    const updatedClaims = await adminService.updateClaims(
+      user.externalId,
+      role.id,
+      !checked
+    )
+    // Then update with server-side result
+    setChecked(updatedClaims.claims.includes(role.id))
+  }, [checked])
+
+  return (
+    <Checkbox label={role.label} checked={checked} onChange={updateClaim} />
+  )
+}
+
+export default function AdminUsersPage() {
+  const { t, lang } = useTranslation('admin')
+  const [search, setSearch] = useState('')
+  const [searchDebounced, setSearchDebounced] = useState('')
+
+  const { page, setPage, handleTableHeaderClick, sortBy, sortDirection } =
+    usePagination<UserSortField>(1, UserSortField.CreatedAt)
+
+  const rolesArray = useMemo<Role[]>(
+    () =>
+      Object.values(FirebaseClaim).map((id) => ({
+        id,
+        label: t(`users.roles.${id}`),
+      })),
+    []
+  )
+
+  const qp = getUsersFilterQuery({
+    page,
+    sortBy,
+    sortDirection,
+    pageSize: USERS_PER_PAGE,
+    search: searchDebounced,
+  })
+  const { data, isValidating } = useAuthApi<UserAccounts>(
+    `${urls.api.v1.admin.getUsers}?${qp}`
+  )
+  const users = data?.users || []
+  const total = data?.total || 0
+
+  const renderClaims = useCallback(
+    ({ item }: RendererProps) =>
+      rolesArray.map((role) => (
+        <CheckRole
+          key={`${item.externalId}-${role.id}`}
+          user={item}
+          role={role}
+        />
+      )),
+    []
+  )
+
+  const submitSearch = useCallback(
+    debounce((value) => {
+      setSearchDebounced(value)
+    }, 200),
+    []
+  )
```

Comment:
The search works great currently, but wondering if it's worth adding a new library. We could wait until the search term is submitted, or possibly build a similar utility ourselves. Curious how others feel though.

Reply:
Cool cool, I'll follow y'all's lead. With `lodash.get` in there, would we be interested in just `lodash` - since I'm adding `lodash.debounce`? Lotsa handies in Lodash. And we'd just double-ensure tree-shaking (eg `npm i lodash-es`, or `import x from 'lodash/x'`) so no added bloat

Reply:
If we can ensure lodash doesn't bloat our JS bundle too much I'm okay with adding _some_ utilities from there. Check output sizes for the web project before and after referencing it.

From my experience though, lodash haven't always played well with tree-shaking features and importing just a single module gives you a lot more than you bargained for. There's also a risk a future developer that is less familiar with these concepts imports all of lodash, if we allow it here.

Reply:
@smonn that's an awesome extension! I'll install it asap

Reply:
GTK on destructuring, looks good if direct lodash import. Also verified I wasn't tripping with icons. This plugin will be a boon, glad you pointed it out

![Screen Shot 2022-03-16 at 10 19 44 AM](https://user-images.githubusercontent.com/195202/158649741-14bc3091-0f8e-46e7-9b6a-6e378a41b3db.png)

Reply:
Here's w lodash.get/debounce -> lodash (plain) + tree-shook imports. More than happy to not do this route btw, and just add "search" button